db/main.py

```diff
@@ -0,0 +1,84 @@
+import base64
+from sqlalchemy import create_engine, Table, MetaData, Column, String, exc
+from sqlalchemy.sql import select
+
+
+class MagpieData(object):
+    def __init__(self, url, metadata=None):
+        self.url = url
+        if(metadata is not None):
+            self.rowcount = 1
+            self.metadata = str(metadata._row[0])
+        else:
+            self.rowcount = 0
+            self.metadata = metadata
+
+
+class DbUtils(object):
+    def establish_db_connection(self):
+        # Establish connection
+        connectionString = 'postgres://' + base64.b64decode(self.username) + ':' + base64.b64decode(self.password) + '@' + base64.b64decode(self.hostName) + '/' + self.dbName
+        self.engine = create_engine(connectionString)
+        self.connection = self.engine.connect()
+
+    def __init__(self):
+        credentials = []
+        # Open credentials file. Should be stored as "username:password"
+        filename = 'credentials.txt'
+        try:
+            with open(filename) as f:
+                for line in f:
+                    line = line.split("\n", 1)[0]
+                    credentials.append(line.split(':', 1))
+        except EnvironmentError:
+            raise
+
+        self.username = base64.b64encode(credentials[0][0])
+        self.password = base64.b64encode(credentials[0][1])
+        self.dbName = credentials[1][1]
+        self.tableName = credentials[2][1]
+        self.hostName = base64.b64encode(credentials[3][1])
+        try:
+            self.establish_db_connection()
+        except exc.SQLAlchemyError:
+            raise
+
+        # Columns of table
+        self.columns = ['url', 'metadata']
+
+        # Create Table Object
+        self.mydata = Table(self.tableName, MetaData(),
+                            Column(self.columns[0], String, primary_key=True),
+                            Column(self.columns[1], String), )
+
+    def __del__(self):
+        # Close connection if established
+        if 'self.connection' in locals():
+            self.connection.close()
+
+    def add_value(self, request_url, my_metadata):
+        # Check if URL already exists
+        s = select([self.mydata.c.metadata]).where(self.mydata.c.url == request_url)
```

Comment:
can we also add a last_updated timestamp in UTC. Make sure you update this value on each add.